api/v1/api.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"net/http"
+
+	"github.com/GoogleCloudPlatform/heapster/manager"
+	sinksApi "github.com/GoogleCloudPlatform/heapster/sinks/api"
+	"github.com/emicklei/go-restful"
+)
+
+type Api struct {
+	manager manager.Manager
+}
+
+// Create a new Api to serve from the specified cache.
+func NewApi(m manager.Manager) *Api {
+	return &Api{
+		manager: m,
+	}
+}
+
+// Register the Api on the specified endpoint.
+func (a *Api) Register(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		Path("/api/v1/metric-export").
```

Comment:
I think this is purpose-built for this usecase, I'd leave metrics for a more generic API we build for querying. Worst case, we symlink them in the future.